ecocheck.py

```diff
@@ -0,0 +1,118 @@
+import datetime
+from gofed.lib.pkgdb2client import get_packages
+from gofed.lib.bitbucketclient import get_commits_bitbucket
+from gofed.lib.githubclient import get_commits_github
+from gofed.lib.kojiclient import KojiClient
+
+
+SUPPORTED_REPOSIRORIES = ['github.com', 'bitbucket.org']
+
+
+def repository_info(clone_url):
+	if not clone_url:
+		return []
+	data = clone_url.split('/')
+	if data[2] in SUPPORTED_REPOSIRORIES:
+		repository = {
+			'host': data[2],
+			'owner': data[3],
+			'package': data[4].split('.')[0]
+		}
+	else:
+		logging.error('Unsupported upstream host: %s' % data[2])
+		return []
+	return repository
+
+
+class EcoCheck:
+	def __init__(self):
+		self.input_validated = False
+		self.new_packages = []
+		self.updated_upstream_repos = []
+		self.new_koji_builds = []
+		self.packages = []
+		self.no_prefix_packages = []
+		self.date = ""
+
+	def _validateInput(self, data):
+		validator = SchemaValidator()
+		schema = "%s/ecocheck_input.json" % getScriptDir(__file__)
+		self.input_validated = validator.validateFromFile(schema, data)
+		return self.input_validated
+
+	def setData(self, data):
+		self.input_validated = False
+		if not self._validateInput(data):
+			return []
+		self.date = data['last_change_date']
+		self.no_prefix_packages = data['no_prefix_packages']
+
+		return True
+
+	def _generateGolangProjectUpdatedPackages(self):
+		data = {}
+		data['last_change_date'] = self.date
+		data['new_packages'] = self.new_packages
+		data['updated_upstream_repositories'] = self.updated_upstream_repos
+		data['new_koji_builds'] = self.new_koji_builds
+
+		return data
+
+	def getData(self):
+		if not self.input_validated:
+			return []
+
+		data = self._generateGolangProjectUpdatedPackages()
+		validator = SchemaValidator()
+		schema = "%s/updated-packages.json" % getScriptDir(__file__)
+		if not validator.validateFromFile(schema, data):
+			logging.error('golang-project-updated-packages is not valid')
+			return {}
+
+		return data
+
+	def execute(self):
+		self.no_prefix_packages.append('golang*')
+		self.packages = []
+		for packagename in self.no_prefix_packages:
+			self.packages = get_packages(packagename)
+		file = open('unresolved_packages.txt', 'w')
+
+		for package in self.packages:
+			if package['upstream_url'] == "":
+				logging.error('Failed to get upstream_url for package ' + package['name'])
+				continue
+			new_commits = {'repo': package['name']}
+
+			#get recent commits
+			repinfo = repository_info(package['upstream_url'])
+			if not repinfo:
+				continue
+			pkg_full_name = repinfo['owner'] + '/' + repinfo['package']
+			if repinfo['host'] == 'github.com':
+				pass
+				new_commits['commit'] = get_commits_github(pkg_full_name,datetime.datetime.strptime(self.date,'%Y-%m-%d'))
+			else:
+				new_commits['commit'] = get_commits_bitbucket(pkg_full_name ,self.date)
+
+			if 'commit' in new_commits and new_commits['commit']:
+				self.updated_upstream_repos.append(new_commits)
+
+			# get recent builds
```

Comment:
This should be independent of the remaining two checks/scans. The ``execute`` method should get a list of all new builds independent of what pkgdb returns. I would suggest to implement three methods:

1) one for retrieving a list of new packages
2) one for retrieving a list of new commits
3) one for retrieving a list of new builds

Execute would combine 1) with a list of current packages. The combines list of package would then be provided to 2) and 3). Both methods could be run in parallel.